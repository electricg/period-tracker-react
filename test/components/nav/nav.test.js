import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import Nav, { NavItem } from '../../../src/components/nav';

describe('NavItem', () => {
  test('should fail returning nothing if no id is passed', () => {
    const component = renderer.create(
      <NavItem path="something" label="something" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should fail returning nothing if no path is passed', () => {
    const component = renderer.create(
      <NavItem id="something" label="something" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should fail returning nothing if no label is passed', () => {
    const component = renderer.create(
      <NavItem id="something" path="something" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Nav', () => {
  test('should succeed with home page', () => {
    const component = renderer.create(
      <StaticRouter location="/" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should fail 404 with not existing page', () => {
    const component = renderer.create(
      <StaticRouter location="/nope" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should succeed with calendar page', () => {
    const component = renderer.create(
      <StaticRouter location="/calendar" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should succeed with calendar sub page', () => {
    const component = renderer.create(
      <StaticRouter location="/calendar/2018" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should succeed with log page', () => {
    const component = renderer.create(
      <StaticRouter location="/log" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should fail 404 with log sub page', () => {
    const component = renderer.create(
      <StaticRouter location="/log/sub" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should succeed with settings page', () => {
    const component = renderer.create(
      <StaticRouter location="/settings" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should fail 404 with settings sub page', () => {
    const component = renderer.create(
      <StaticRouter location="/settings/sub" context={{}}>
        <Nav />
      </StaticRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
