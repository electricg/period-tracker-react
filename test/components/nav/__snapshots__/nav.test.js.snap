// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should succeed with calendar page 1`] = `
<nav
  className="main-nav"
>
  <ul
    className="main-nav__list"
  >
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-home"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          home
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current="page"
        className="main-nav__link main-nav__link--selected"
        href="/calendar"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-calendar"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          calendar
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/log"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-log"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          log
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/settings"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-settings"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          settings
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`should succeed with calendar sub page 1`] = `
<nav
  className="main-nav"
>
  <ul
    className="main-nav__list"
  >
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-home"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          home
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current="page"
        className="main-nav__link main-nav__link--selected"
        href="/calendar"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-calendar"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          calendar
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/log"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-log"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          log
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/settings"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-settings"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          settings
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`should succeed with home page 1`] = `
<nav
  className="main-nav"
>
  <ul
    className="main-nav__list"
  >
    <li
      className="main-nav__item"
    >
      <a
        aria-current="page"
        className="main-nav__link main-nav__link--selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-home"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          home
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/calendar"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-calendar"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          calendar
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/log"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-log"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          log
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/settings"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-settings"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          settings
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`should succeed with log page 1`] = `
<nav
  className="main-nav"
>
  <ul
    className="main-nav__list"
  >
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-home"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          home
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/calendar"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-calendar"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          calendar
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current="page"
        className="main-nav__link main-nav__link--selected"
        href="/log"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-log"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          log
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/settings"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-settings"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          settings
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`should succeed with settings page 1`] = `
<nav
  className="main-nav"
>
  <ul
    className="main-nav__list"
  >
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-home"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          home
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/calendar"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-calendar"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          calendar
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current={null}
        className="main-nav__link"
        href="/log"
        onClick={[Function]}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-log"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          log
        </span>
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        aria-current="page"
        className="main-nav__link main-nav__link--selected"
        href="/settings"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="icon icon--main-nav"
          role="presentation"
        >
          <use
            xlinkHref="#icon-settings"
          />
        </svg>
        <span
          className="main-nav__link-text"
        >
          settings
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
